probe_src/python/probe_py/manual/cli.py

```diff
@@ -146,5 +146,63 @@ def dump(
                     print(pid, exid, tid, op_no, op.data)
                 print()
 
+@app.command(
+    context_settings=dict(
+        ignore_unknown_options=True,
+    ),
+)
+def mpirun(
+        cmd: list[str],
+        np: int = typer.Option(default=1, help="Number of processes to use in mpirun"),
+        gdb: bool = typer.Option(default=False, help="Run in GDB"),
+        debug: bool = typer.Option(default=False, help="Run verbose & debug build of libprobe"),
+        make: bool = typer.Option(default=False, help="Run make prior to executing"),
+        output: pathlib.Path = pathlib.Path("probe_log"),
+        no_transcribe: bool = typer.Option(default=False, help="Only execute without transcribing"),
+) -> None:
+    """
+    Wrap mpirun to execute programs on N remote machines and record its provenance.
```

Comment:
Part of the task is to determine if we need a wrapper or will `mpirun probe record ./foobar` work? My concern is, in the future, when we track multli-node communication: will we be able to determine that communication occurs by examining the system calls of `./foobar`? or will we have to intercept `mpirun` and say "ok, communication is occurring between this set of nodes; now go run `./foobar`"?